### DISCUSSION TITLE
Linking against Homebrew-provided libs in Linux

### DISCUSSION THREAD

**Discussion by native-api** [Homebrew/discussions#2945] [Linux] [answered]
Pyenv users are having trouble building Pythons in Linux built against Homebrew-provided libs (see https://github.com/pyenv/pyenv/issues/2159).
The build runs fine but the libs are not founf at runtime: https://github.com/pyenv/pyenv/issues/2159#issuecomment-968303037.

For some reason, it works if one uses the GCC provided by Homebrew.

Do we need to pass some additional compiler flags? Is there any guideline on the recommended way to link against Homebrew-provided libs?

**Comment by carlocab** [comment, marked as answer]
Homebrew GCC is configured to add the correct RPATHs. We may add this to `.pc` files we ship too so that it works with `pkg-config` and a non-Homebrew GCC too.

If you want to avoid the need for a Homebrew GCC, adding
```
LDFLAGS="-Wl,-rpath,$(brew --prefix)/lib"
```
should help.
